indico/modules/events/editing/client/js/editing/timeline/TimelineItem.jsx

```diff
@@ -43,24 +43,27 @@ export default function TimelineItem({block, index}) {
   const {fileTypes} = useSelector(selectors.getStaticData);
   const isLastBlock = lastTimelineBlock.id === block.id;
   const isLastValidBlock = lastValidTimelineBlock.id === block.id;
-  const [visible, setVisible] = useState(isLastValidBlock);
-  const isUndone = block.finalState.name === FinalRevisionState.undone;
+  const hasContent = block.commentHtml || !!block.files.length;
+  const isAlwaysVisible = isLastValidBlock || lastTimelineBlockWithFiles.id === block.id;
+  const [visible, setVisible] = useState(isAlwaysVisible);
 
   useEffect(() => {
```

Comment:
Would it be simpler if instead of the useEffect, you check `isAlwaysVisible` when setting the visibility?
Something like this:
```js
onClick={() => {
  if(!visible || !isAlwaysVisible) { setVisible(v => !v) }
}}
```

Reply:
found an even simpler solution:
```js
const [visibilityToggle, setVisibilityToggle] = useState(false);
const visible = visibilityToggle || isAlwaysVisible;
```